src/iqstat.cpp

```diff
@@ -363,3 +241,94 @@ void usage() {
+                    const char *ruleName, 
+                    int allFlag, 
+                    bool brief ) -> int
+{
+    genQueryInp_t genQueryInp{};
+    genQueryOut_t *genQueryOut{};
+
+    auto num_cols_selected = (brief ? once_instance.brief_format_len
+                                    : once_instance.long_format_len);
+
+    for (int i = 0; i < num_cols_selected; i++) {
+        addInxIval(&genQueryInp.selectInp,i1a[i],0);
+    }
+
+    if (!allFlag) {
+        snprintf( v1, BIG_STR, "='%s'", userName );
+        addInxVal(&genQueryInp.sqlCondInp,COL_RULE_EXEC_USER_NAME,v1);
+    }
+
+    std::string diagnostic {""};
+    if (ruleName != NULL && *ruleName != '\0') {
+        diagnostic = (boost::format( " (and matching key '%s')" ) % ruleName).str();
+        snprintf( v2, BIG_STR, "='%s'", ruleName );
+        addInxVal(&genQueryInp.sqlCondInp,COL_RULE_EXEC_ID,v2);
+    }
+
+    genQueryInp.maxRows = 50;
+    status = rcGenQuery( Conn, &genQueryInp, &genQueryOut );
+
+    if ( status == CAT_NO_ROWS_FOUND ) {
+        // Need to determine which table "no rows" refers to: USER or RULE.
+        namespace ia = irods::experimental::administration;
+        if (ia::client::exists(*Conn, ia::user{userName})) {
+            if ( allFlag ) {
+                printf( "No delayed rules pending%s\n", diagnostic.c_str() );
+            }
+            else {
+                printf( "No delayed rules pending for user %s%s\n", userName, diagnostic.c_str());
+            }
+            return 0;
+        }
+        else {
+            printf( "User %s does not exist.\n", userName );
+            return 0;
+        }
+    }
+
+    if (brief) {
+        printf( "id     name\n" );
+    }
+
+    auto increment_count_and_print_results = [&] {
+       printCount += printGenQueryResults( Conn, status, genQueryOut,
+                                           brief ? NULL: columnNames,
+                                           int{brief} );
+    };
+
+    increment_count_and_print_results();
+
+    while ( status == 0 && genQueryOut->continueInx > 0 ) {
+        genQueryInp.continueInx = genQueryOut->continueInx;
+        status = rcGenQuery( Conn, &genQueryInp, &genQueryOut );
+        if (!brief && genQueryOut->rowCnt > 0 ) {
+            printf( "----\n" );
+        }
+        increment_count_and_print_results();
+        clearGenQueryOut( genQueryOut );
+    }
+    clearGenQueryInp( &genQueryInp );
+    return 0;
+}
+
+auto genQuery_id_string_to_ulong(const std::string & key) -> unsigned long
+{
+    std::string k{key};
+    boost::algorithm::trim(k);
+    try {
+        return std::stoul(k);
+    }
+    catch(const std::invalid_argument&){
+        THROW(SYS_INVALID_INPUT_PARAM, "Delayed task ID has incorrect format.");
+    }
+    catch(const std::out_of_range&){
+        THROW(SYS_INVALID_INPUT_PARAM, "Delayed task ID is too large.");
+    }
+    catch(...){
+        THROW(SYS_INVALID_INPUT_PARAM, "Unknown error parsing task ID.");
```

Comment:
Let's replace `task` with `rule` since that is the most common terminology used to describe a delay rule.

Reply:
Ok. Change "Delayed task" to "Delay rule" on both lines then.  LMK if something else occurs that is better.